### PR TITLE
added cramer von-mises two sample test

### PR DESCRIPTION
**Pull request by astronomeralex**
I've added the cramer von-mises two sample test to stats. This is my first pull request, so I'm sure there's something that I didn't do quite right in the documentation.

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -3900,6 +3900,46 @@ def ks_2samp(data1, data2):
...
+    ri = allranks[:N]
+    sj = allranks[-M:]
+    i = rankdata(x)
+    j = rankdata(y)
+    # Anderson et al. Eqn 10
+    U = N*np.sum((ri - i)**2) + M*np.sum((sj - j)**2)
+    # Anderson et al. Eqn 9
+    T = U/(N * M * (N + M)) - (4 * M * N - 1)/(6 * (M + N))
+    Texpected = 1/6 + 1/(6 * (M + N))
+    Tvariance = 1/45 * (M + N + 1)/(M + N)**2 * (4 * M * N * (M+N) - \
```
Comment (dankosaur) [inline scipy/stats/stats.py]: `Texpected` and `Tvariance` come out consistently as zero due to the fact that the math here is integer-based.  Need to make these two lines use floats.

Comment (pv) [inline scipy/stats/stats.py]: Integer true division is active in this file also on Python 2, so this should be OK.

**Comment by rgommers** [comment]
No follow-up for several years, and gh-7563 has a Cramer - Von Mises distance that looks more complete, so closing.

**Comment by josef-pkt** [comment]
The new distance pr doesn't cover this because it doesn't have the test statistic

**Comment by rgommers** [comment]
You're right, thanks for the correction Josef. Still leaving this closed - no action in 3 years and not high prio. Tagged with needs-champion if we're ever getting to the point where we have more bandwidth than active PRs:)

**Pull request closed without merging**